logback-access/src/main/java/ch/qos/logback/access/spi/AccessEvent.java

```diff
@@ -335,13 +335,22 @@ public void buildRequestHeaderMap() {
 
     public void buildRequestParameterMap() {
         requestParameterMap = new HashMap<String, String[]>();
-        Enumeration<String> e = httpRequest.getParameterNames();
-        if (e == null) {
-            return;
-        }
-        while (e.hasMoreElements()) {
-            String key = e.nextElement();
-            requestParameterMap.put(key, httpRequest.getParameterValues(key));
+        try {
+            Enumeration<String> e = httpRequest.getParameterNames();
+            if (e == null) {
+                return;
+            }
+            while (e.hasMoreElements()) {
+                String key = e.nextElement();
+                requestParameterMap.put(key, httpRequest.getParameterValues(key));
+            }
+        } catch(Throwable t) {
```

Comment:
The Servlet Spec doesn't specify the exceptions that can happen here.
It can be something as simple as an IOException (which is a checked exception) that has to be wrapped in something else.
It can be a parsing issue, it can be a Socket issue, it can be charset issue, etc...

Unlike a normal webapp and how it handles failure, a RequestLog cannot skip its logging.
Ideally, a RequestLog should never call a method that changes the Request or Response state.

Reply:
Here's a few things that can happen at the `Error` level within the Servlet that are non-fatal for the JVM or ServletContext ...

* XML initialization errors (stream, parser, validation, etc) (endpoint specific)
* java.io.IOError with non-socket based connections/connectors (like UnixSocket)
* ServiceLoader and Service provider errors. (endpoint specific)
* java.nio.Charset encode/decode errors. (request specific)

Reply:
The logging here in RequestLog should not fail for a fault out of its control.

Reply:
@joakime is this implementation still correct? I've tested Logback 1.3.0-alpha14 (which basically uses the same implementation without the try block) with Jetty 11.0.8 and at this method a `NullPointerException` gets thrown from [here](https://github.com/eclipse/jetty.project/blob/jetty-11.0.x/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java#L478).